File: DragonsGenerator.API/Models/Deity.cs
namespace DragonsGenerator.API.Models;

public record Deity(
    string Id,
    string Name,
    string? Tonality,
    List<string> Domains,
    string? Description,
    List<string> OtherNames,
    string? WorshippersNote,
    List<string> GrantsPowersTo,
    string? Source = null
);

File: DragonsGenerator.WEB/src/app/core/models/Deities/deity.ts
export interface Deity {
  id: string;
  name: string;
  tonality?: string | null;
  domains: string[];
  description?: string | null;
  otherNames: string[];
  worshippersNote?: string | null;
  grantsPowersTo: string[];
  source?: string | null;
}

export interface DeitySummary {
  id: string;
  name: string;
  tonality?: string | null;
  domains: string[];
}
